nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestConvertRecord.java

```diff
@@ -434,7 +435,10 @@ public void testJSONDroppingUnknownFields() throws InitializationException, IOEx
         runner.assertAllFlowFilesTransferred(ConvertRecord.REL_SUCCESS, 1);
 
         final MockFlowFile flowFile = runner.getFlowFilesForRelationship(ConvertRecord.REL_SUCCESS).getFirst();
-        assertFalse(flowFile.getContent().contains("fieldThatShouldBeRemoved"));
+
+        // This covers all the cases
+        // "undefinedKey", "undefinedObjectArray", "undefinedObject", "undefinedScalarArray"
```

Comment:
Thanks for the updates @dan-s1 but this list is missing "undefinedKeyInObject".

Reply:
@mosermw Thanks for pointing that out.